app/src/main/java/com/lagradost/cloudstream3/ui/settings/SettingsFragment.kt

```diff
@@ -136,21 +138,44 @@ class SettingsFragment : Fragment() {
             activity?.navigate(id, Bundle())
         }
 
-        // used to debug leaks showToast(activity,"${VideoDownloadManager.downloadStatusEvent.size} : ${VideoDownloadManager.downloadProgressEvent.size}")
-
-        for (syncApi in accountManagers) {
-            val login = syncApi.loginInfo()
-            val pic = login?.profilePicture ?: continue
-            if (binding?.settingsProfilePic?.setImage(
-                    pic,
-                    errorImageDrawable = HomeFragment.errorProfilePic
-                ) == true
-            ) {
-                binding?.settingsProfileText?.text = login.name
-                binding?.settingsProfile?.isVisible = true
-                break
+        /** used to debug leaks
+        showToast(activity,"${VideoDownloadManager.downloadStatusEvent.size} :
+        ${VideoDownloadManager.downloadProgressEvent.size}") **/
+
+        // Check login status for each OAuth2API
+        val accountProfile = accountManagers.firstOrNull { it.loginInfo() != null }
```

Comment:
in this case it will load pfp of the local account. and name also ?

Reply:
no, lets say you have a non null loginInfo, but that loginInfo does not have a pfp then we are fucked

Reply:
i am not understanding, the login info

![image](https://github.com/recloudstream/cloudstream/assets/125901294/03a887d6-7e55-42f0-ab46-b4cdd839a35d)

the loginInfo() returns null when NOT logged in, and we are already using that form

![image](https://github.com/recloudstream/cloudstream/assets/125901294/51d36c8f-5e0b-4647-a90a-5161626aceb1)

?

Reply:
assume that accountmanagers is a list of 

```
listof(LoginInfo(
pfp = null,
name = "Hello",
accountIndex = 0
),
LoginInfo(
pfp = null,
name = "World",
accountIndex = 1
))
```
What would your code do? It would NOT work as it passes the accountProfile check (because it has more than one account), but does not in fact have an image, and because of this `val pic = login?.profilePicture ?: continue` will be trigged for both and you will get NOTHING.